### PR TITLE
Minor refactoring to process_payment

### PR DESCRIPTION
**Pull request by jessy-p**
#### Changes proposed in this Pull Request
Moved few lines of code into methods as a first step towards Re-engineering Payments Process project

#### Testing instructions

* Tests pass
* Able to checkout

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6459` or branch name `refactor/process-payment-readability` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=refactor/process-payment-readability) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 898efe01898468b01b15d356e0f926ebaf33c2a9
- Build time: 2023-06-05 12:42:25 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payment-gateway-wcpay.php [hunk 1]
```diff
@@ -856,6 +838,55 @@ public function process_payment( $order_id ) {
...
 	}
 
+	/**
+	 * Verifies the phone number to be within valid length.
+	 *
+	 * @param string $phone_number Phone number to be validated.
+	 *
+	 * @throws Process_Payment_Exception
+	 */
+	private function verify_phone_number( string $phone_number ) {
```
Comment (naman03malhotra) [inline includes/class-wc-payment-gateway-wcpay.php:848]: Do we write test for this change in this PR or the next?

**Review by naman03malhotra — COMMENTED** [review]
Code refactor looks good to me, left a comment about tests.
I haven't tested it locally though.

File: includes/class-wc-payment-gateway-wcpay.php (hunk 1 above)
Comment (jessy-p) [reply includes/class-wc-payment-gateway-wcpay.php:848]: There are already tests for the public function in `WC_Payment_Gateway_WCPay_Process_Payment_Test` which tests these lines that verify phone number, so I think I don't need to introduce any tests for the new private function.

Comment (jessy-p) [reply includes/class-wc-payment-gateway-wcpay.php:848]: I can introduce tests as part of the Payments_Service, later.

**Review by naman03malhotra — APPROVED** [review]
Looks good to me, but I recommend someone to test it as well as I couldn't.

**Comment by jessy-p** [comment]
Closing as not required anymore

**Pull request closed without merging**